test.js

```diff
@@ -148,14 +148,47 @@ function testStep(context, conversationId, userMessage, expectedReplies, timeout
         });
 }
 
+// Replacement method for chai's deep.equal
+function deepEqual(expected, actual) {
+    // Test using regex for attributes starting with regex keyword 
+    if ((typeof expected === "string" && expected.startsWith("regex:"))) {
+        const regex = new RegExp(expected.replace("regex:", "").trim());
+        if (!regex.test(actual))
+            throw "Actual value doesn't match provided regex, Regex: " + regex.toString() + ", Actual: " + JSON.stringify(actual);
+        else return true;
+    }
+    // Regular test for other values
+    else if (expected === actual)
+        return true;
+    else if ((typeof expected === "object" && expected !== null) && (typeof actual === "object" && actual !== null)) {
+        for (var prop in expected) {
+            if (actual.hasOwnProperty(prop)) {
+                try {
+                    deepEqual(expected[prop], actual[prop]);
+                } catch (error) {
+                    if (error === "1")
+                        throw "Cards are not equal, Error in attribute '" + prop + "', expectedValue: " + JSON.stringify(expected[prop]) + ", actualValue: " + actual[prop];
+                    throw error;
+                }
+            } else {
+                throw "Actual card is missing '" + prop + "' property";
+            }
+        }
+        return true;
+    }
+    else {
+        throw "1";

```

Comment:
Maybe throw a string with message instead of "1".

Reply:
hmm is "Attributes mismatch" good?

Reply:
I agree, is "Attributes mismatch" good enough❓

Reply:
Yes, it's fine.